File: solutions/julia/log-levels/1/log-levels.jl
function message(msg)
    strip(split(msg, ":")[2])
end

function log_level(msg)
    raw_level = split(msg, ":")[1]
    upper_level = replace(raw_level, "["=>"", "]"=>"")
    lower_level = lowercase(upper_level)
    return lower_level
end

function reformat(msg)
    message(msg) * " (" * log_level(msg) * ")"
end

File: solutions/r/log-levels/1/log-levels.R
library(stringr)

message <- function(msg) {
  trimws(unlist(strsplit(msg, ":"))[2])
}

log_level <- function(msg) {
  log <- unlist(strsplit(msg, ":"))[1]
  start <- 2
  end <- nchar(log)-1
  str_to_lower(substr(log, start, end))
}

reformat <- function(msg) {
  m <- message(msg)
  l <- log_level(msg)
  vl <- c("(", l, ")")
  fl <- str_flatten(vl, "")
  vmfl <- c(m,fl)
  str_flatten(vmfl, " ")
}
